src/librustc_mir/build/matches/mod.rs

```diff
@@ -145,8 +145,18 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
         end_block.unit()
     }
 
+    pub fn user_assert_ty(&mut self, block: BasicBlock, ty: Ty<'tcx>, var: NodeId, span: Span) {
```

Comment:
I've been debating whether my idea to make this a statement was good or not -- but I do see some advantages. For example, it neatly addresses the question of *where* this subtyping needs to be enforced.